components/dashboard/src/user-settings/Preferences.tsx

```diff
@@ -72,6 +100,42 @@ export default function Preferences() {
                         </p>
                     </div>
                 </div>
+
+                <h3 className="mt-12">Inactivity Timeout </h3>
+                <p className="text-base text-gray-500 dark:text-gray-400">
+                    By default, workspaces stop following 30 minutes without user input (e.g. keystrokes or terminal
+                    input commands). You can increase the workspace timeout up to a maximum of 24 hours.
+                </p>
+                <div className="mt-4 max-w-xl">
+                    <h4>Default Inactivity Timeout</h4>
+                    <span className="flex">
+                        <input
+                            type="text"
+                            className="w-96 h-9"
+                            value={workspaceTimeout}
+                            placeholder="timeout time, such as 30m, 1h, max 24h"
+                            onChange={(e) => setWorkspaceTimeout(e.target.value)}
+                        />
+                        <button
+                            className="secondary ml-2"
+                            onClick={() => actuallySetWorkspaceTimeout(workspaceTimeout)}
+                        >
+                            Save Changes
+                        </button>
+                    </span>
+                    <div className="mt-1">
+                        <p className="text-gray-500 dark:text-gray-400">
+                            Use minutes or hours, like <strong>30m</strong> or <strong>2h</strong>.
+                        </p>
+                    </div>
+
+                    <CheckBox
+                        title="Keep workspace running"
```

Comment:
Keep workspace running seems quite ambiguous to me. 

Could we be more explicit? 

```suggestion
                        title="Stop workspace if no active editor connection"
```